### PR TITLE
Fixed Doublechests in RegenTask, Added Ally methods & PlayerEnter- Leave TownEvents

### PR DESCRIPTION
**Pull request by Articdive**
Title

### CODE REVIEW

**Comment by Articdive** [comment]
Closes #2707

File: src/com/palmergames/bukkit/towny/event/PlayerEnterTownEvent.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+		this.from = from;
+		this.pme = pme;
+		this.to = to;
+	}
+
+	public Player getPlayer() {
+		return player;
+	}
+
+	public PlayerMoveEvent getPme() {
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/event/PlayerEnterTownEvent.java]: This should be changed to getMoveEvent() for clarity.

File: src/com/palmergames/bukkit/towny/event/PlayerLeaveTownEvent.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		this.from = from;
+		this.pme = pme;
+		this.to = to;
+	}
+
+	public Player getPlayer() {
+		return player;
+	}
+
+	public PlayerMoveEvent getPme() {
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/event/PlayerLeaveTownEvent.java]: Same here, change to getPlayerMoveEvent()

**Comment by LlmDl** [comment]
Added a couple comments, and the german.yml needs resolving, otherwise it's good.

**Pull request merged**